packages/next-on-pages/tests/templates/utils/http.test.ts

```diff
@@ -81,6 +81,61 @@ describe('applySearchParams', () => {
 			'http://localhost/page?other=value&foo=bar',
 		);
 	});
+
+	test('allows multiple query params with the same key', () => {
+		const source = new URL('http://localhost/page?foo=bar');
+		const target = new URL(
+			'http://localhost/page?other=value&foo=baz&foo=test',
+		);
+
+		expect([...source.searchParams.entries()].length).toEqual(1);
+		expect([...target.searchParams.entries()].length).toEqual(3);
+
+		applySearchParams(target.searchParams, source.searchParams);
+
+		expect([...source.searchParams.entries()].length).toEqual(1);
+		expect([...target.searchParams.entries()].length).toEqual(4);
```

Comment:
I'm confused why we need to do this and can't just do `.size`, I did try it locally and `.size` is `undefined`... do you have any idea as to why? is this a weird vitest thing? 😕

Reply:
is this related? 🤔  https://github.com/vercel/next.js/issues/49245

Reply:
Seems like `size` has only recently been added in V8 so maybe workerd is simply lagging a bit behind 🤷 

I'm totally happy with this version then 🙂